pkg/connector/instance.go

```diff
@@ -86,7 +86,7 @@ type Connector interface {
 
 // PluginDispenserFetcher can fetch a plugin dispenser.
 type PluginDispenserFetcher interface {
-	NewDispenser(logger log.CtxLogger, name string) (connectorPlugin.Dispenser, error)
+	NewDispenser(logger log.CtxLogger, name string, connectorID string) (connectorPlugin.Dispenser, error)
```

Comment:
Random thing I noticed: a dispenser is able to dispense sources and destinations, but in practice a dispenser dispenses either sources or destinations.